src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
+            foreach ($uuidAttributes as $attributeKey) {
+                if (array_key_exists($attributeKey, $array)) {
+                    $uuidKey = $this->getRelatedBinaryKeyName($attributeKey);
+                    $array[$attributeKey] = $this->{$uuidKey};
+                }
+            }
+        }
+
+        return $array;
+    }
+
+    public function getRelatedBinaryKeyName($attrib)
+    {
+        $suffix = $this->getUuidTextAttributeSuffix();
+
+        return "{$attrib}{$suffix}";
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
+            return static::decodeUuid($this->{$uuidKey});
+        }
+
+        return parent::getAttribute($key);
+    }
+
+    public function setAttribute($key, $value)
+    {
+        if ($uuidKey = $this->uuidTextAttribute($key)) {
+            $value = static::encodeUuid($value);
         }
 
-        $data = parent::toArray();
+        return parent::setAttribute($key, $value);
+    }
+
+    protected function getUuidTextAttributeSuffix()
```

Comment:
Maybe we should consider a more eloquent name here? Something like `getTextualUuidSuffix` or even `getUuidSuffix`? What's your opinion?

Reply:
`getUuidSuffix` seems better... i always like to go for the shortest. I was at work and needed to finish early so the short names weren't popping up in my head fast... Lol